src/artm/core/processor.cc

```diff
@@ -469,13 +537,19 @@ InferThetaAndUpdateNwtSparse(const ModelConfig& model_config, const Batch& batch
     for (float& value : values) value *= batch_weight;
     nwt_writer->Store(w, token_id[w], values);
   }
+
+  CreateThetaCacheEntry(new_cache_entry_ptr, theta_matrix, batch, part->model_name(), model_increment, topics_count);
 }
 
 static void
 InferPtdwAndUpdateNwtSparse(const ModelConfig& model_config, const Batch& batch, float batch_weight, const Mask* mask,
                             const InstanceSchema& schema, const CsrMatrix<float>& sparse_ndw,
                             const ::artm::core::PhiMatrix& p_wt, DenseMatrix<float>* theta_matrix,
-                            NwtWriteAdapter* nwt_writer, util::Blas* blas) {
+                            NwtWriteAdapter* nwt_writer, util::Blas* blas,
+                            DataLoaderCacheEntry* new_cache_entry_ptr = nullptr,
+                            DataLoaderCacheEntry* new_ptdw_cache_entry_ptr = nullptr,
+                            ModelIncrement* model_increment = nullptr,
```

Comment:
Would it be possible to avoid passing model_increment and part? I think they are spreading across the code for a  very minor reason --- you use them for in CreateThetaCacheEntry to do this:

+  new_cache_entry_ptr->set_batch_uuid(batch.id());
+  new_cache_entry_ptr->set_model_name(model_name);
+  new_cache_entry_ptr->mutable_topic_name()->CopyFrom(model_increment->topic_model().topic_name());

I would rather move this back into ThreadFunction...